Add bebop ip address as ROS parameter

anuppari opened the pull request:
I added the IP address of the bebop as a ROS parameter. This makes it easier to run when the bebop is connected as a client to a separate AP, e.g., for running multiple bebop simultaneously.

anuppari committed:
Add bebop ip address as ROS parameter

mani-monaj commented on bebop_driver/include/bebop_driver/bebop.h:
```diff
@@ -138,7 +139,7 @@ class Bebop
   inline bool IsStreamingStarted() const {return is_streaming_started_;}
 
-  explicit Bebop(ARSAL_Print_Callback_t custom_print_callback = 0);
+  explicit Bebop(ARSAL_Print_Callback_t custom_print_callback = 0, const std::string& = "192.168.42.1");
```
Please include the argument name.

mani-monaj commented on bebop_driver/launch/bebop_node.launch:
```diff
@@ -3,6 +3,7 @@
     <group ns="bebop">
         <node pkg="bebop_driver" name="bebop_driver" type="bebop_driver_node" output="screen">
             <param name="camera_info_url" value="package://bebop_driver/data/bebop_camera_calib.yaml" />
+            <param name="bebop_ip" value="192.168.0.101" />
```
Please use the default IP address of Bebop: `192.168.42.1`

mani-monaj commented on bebop_driver/src/bebop_driver_nodelet.cpp:
```diff
@@ -77,8 +76,11 @@ void BebopDriverNodelet::onInit()
   // TODO(mani-monaj): Wrap all calls to .param() in a function call to enable logging
   const bool param_reset_settings = private_nh.param("reset_settings", false);
   const std::string& param_camera_info_url = private_nh.param<std::string>("camera_info_url", "");
+  const std::string& bebop_ip = private_nh.param<std::string>("bebop_ip", "192.168.42.1");
```
Please refactor  `bebop_ip` to `param_bebop_ip`

mani-monaj replied:
I mean the variable name, not the ROS parameter.

mani-monaj commented on bebop_driver/src/bebop_driver_nodelet.cpp:
```diff
@@ -77,8 +76,11 @@ void BebopDriverNodelet::onInit()
 
   param_frame_id_ = private_nh.param<std::string>("camera_frame_id", "camera");
+  
+  bebop_ptr_.reset(new bebop_driver::Bebop(util::BebopPrintToROSLogCB,bebop_ip));
```
Space before `bebop_ip` is missing.

mani-monaj replied:
Please use `boost::make_shared` instead of `reset`.

mani-monaj replied:
This assumes that we do not agree on this: https://github.com/AutonomyLab/bebop_autonomy/pull/19/files#r45271943

mani-monaj commented on bebop_driver/src/bebop_driver_nodelet.cpp:
```diff
@@ -59,7 +59,6 @@ int BebopPrintToROSLogCB(eARSAL_PRINT_LEVEL level, const char *tag, const char *
 }  // namespace util
 
 BebopDriverNodelet::BebopDriverNodelet()
-  : bebop_ptr_(new bebop_driver::Bebop(util::BebopPrintToROSLogCB))
```
This is a bit dangerous, since now there is no guarantee that this pointer is not NULL and the object is initialized when for example a callback is called when `onInit()` method is not finished. I am not sure if that is actually a case when using nodelets, however I like the safety that this early allocation provides. 

I suggest that we leave the original constructor untouched and pass/save `bebop_ip` when calling `Connect()`.

mani-monaj commented:
Thank you for the PR, please see my inline comments. Specifically [this one](https://github.com/AutonomyLab/bebop_autonomy/pull/19/files#r45271943).

anuppari committed:
Change initialization of bebop ip
- Reset bebop initialization based on https://github.com/AutonomyLab/bebop_autonomy/pull/19/files#r45271943
- Changed bebop parameter variable name
- Set default ip in launch files

anuppari commented:
OK I made those changes you recommended. Do I need to make a new PR?

mani-monaj commented:
@anuppari Great. Thank you. It looks good to me. It would be nice if you can [squash](http://eli.thegreenplace.net/2014/02/19/squashing-github-pull-requests-into-a-single-commit) your commits into one and update the PR. I will merge this quite soon.

anuppari commented:
Sorry it took a while, but I made a new PR with the squashed commits

mani-monaj commented:
No worries, I've already squashed and merged it.

dvalenciar commented:
can anyone confirm that this change in the code works well?
